onmt/model_builder.py

```diff
@@ -157,62 +150,52 @@ def build_base_model(model_opt, fields, gpu, checkpoint=None):
-        encoder = build_encoder(model_opt, src_embeddings)
+        feat_fields = [fields[k]
+                       for k in inputters.collect_features(fields, 'src')]
+        src_emb = build_embeddings(model_opt, fields["src"], feat_fields)
+        encoder = build_encoder(model_opt, src_emb)
     elif model_opt.model_type == "img":
-        if ("image_channel_size" not in model_opt.__dict__):
+        # why is the model_opt.__dict__ check necessary?
+        if "image_channel_size" not in model_opt.__dict__:
             image_channel_size = 3
         else:
             image_channel_size = model_opt.image_channel_size
 
-        encoder = ImageEncoder(model_opt.enc_layers,
-                               model_opt.brnn,
-                               model_opt.enc_rnn_size,
-                               model_opt.dropout,
-                               image_channel_size)
+        encoder = ImageEncoder(
+            model_opt.enc_layers, model_opt.brnn, model_opt.enc_rnn_size,
+            model_opt.dropout, image_channel_size)
     elif model_opt.model_type == "audio":
-        encoder = AudioEncoder(model_opt.rnn_type,
-                               model_opt.enc_layers,
-                               model_opt.dec_layers,
-                               model_opt.brnn,
-                               model_opt.enc_rnn_size,
-                               model_opt.dec_rnn_size,
-                               model_opt.audio_enc_pooling,
-                               model_opt.dropout,
-                               model_opt.sample_rate,
-                               model_opt.window_size)
+        encoder = AudioEncoder(
+            model_opt.rnn_type, model_opt.enc_layers, model_opt.dec_layers,
+            model_opt.brnn, model_opt.enc_rnn_size, model_opt.dec_rnn_size,
+            model_opt.audio_enc_pooling, model_opt.dropout,
+            model_opt.sample_rate, model_opt.window_size)
 
     # Build decoder.
-    tgt_dict = fields["tgt"].vocab
-    feature_dicts = inputters.collect_feature_vocabs(fields, 'tgt')
-    tgt_embeddings = build_embeddings(model_opt, tgt_dict,
-                                      feature_dicts, for_encoder=False)
+    feat_fields = [fields[k]
+                   for k in inputters.collect_features(fields, 'tgt')]
+    tgt_emb = build_embeddings(
+        model_opt, fields["tgt"], feat_fields, for_encoder=False)
 
     # Share the embedding matrix - preprocess with share_vocab required.
     if model_opt.share_embeddings:
         # src/tgt vocab should be the same if `-share_vocab` is specified.
-        if src_dict != tgt_dict:
-            raise AssertionError('The `-share_vocab` should be set during '
-                                 'preprocess if you use share_embeddings!')
+        assert fields['src'].vocab == fields['tgt'].vocab, \
+            """-share_vocab should be set during
+            preprocessing if you use share_embeddings"""
```

Comment:
There would be unwanted whitespaces in the AssertError message. 
Close quotes, e.g. something like:
```
        assert fields['src'].vocab == fields['tgt'].vocab, "-share_vocab " \
            "should be set during preprocessing if you use share_embeddings"
```

Reply:
Fix coming. Thanks!